contracts/src/bridge/FastBridgeSender.sol

```diff
@@ -15,31 +15,36 @@ import "./interfaces/IFastBridgeSender.sol";
 import "./interfaces/IFastBridgeReceiver.sol";
 
 contract FastBridgeSender is IFastBridgeSender {
-    ISafeBridge safebridge;
-    IFastBridgeReceiver fastBridgeReceiver;
+    ISafeBridge public safebridge;
+    IFastBridgeReceiver public fastBridgeReceiver;
+    address public fastSender;
 
     /**
      * The bridgers need to watch for these events and
      * relay the messageHash on the FastBridgeReceiver.
      */
     event OutboxMessage(address target, bytes32 messageHash, bytes messagePreImage);
 
-    constructor(ISafeBridge _safebridge, IFastBridgeReceiver _fastBridgeReceiver) {
+    constructor(
+        ISafeBridge _safebridge,
+        IFastBridgeReceiver _fastBridgeReceiver,
+        address _fastSender
+    ) {
         safebridge = _safebridge;
         fastBridgeReceiver = _fastBridgeReceiver;
+        fastSender = _fastSender;
     }
 
     /**
      * Sends an arbitrary message from one domain to another
      * via the fast bridge mechanism
      *
-     * TODO: probably needs some access control either on the sender side
-     * or the receiver side
-     *
      * @param _receiver The L1 contract address who will receive the calldata
      * @param _calldata The receiving domain encoded message data.
      */
     function sendFast(address _receiver, bytes memory _calldata) external {
+        require(msg.sender == fastSender, "Access not allowed: Fast Sender only.");
```

Comment:
The reason I didn't add this initially was to make the fast bridge app-agnostic.